packages/graphiql-toolkit/src/create-fetcher/lib.ts

```diff
@@ -187,9 +210,21 @@ export async function getWsFetcher(
     return createWebsocketsFetcherFromClient(options.wsClient);
   }
   if (options.subscriptionUrl) {
+    const fetcherOptsHeaders = new Headers(fetcherOpts?.headers ?? {});
+    // @ts-expect-error: Current TS Config target does not support `Headers.entries()` method.
+    // However it is reported as "widely available" and so should be fine to use. This could
+    // would be more complicated without it.
+    // @see https://developer.mozilla.org/en-US/docs/Web/API/Headers/entries
+    const fetcherOptsHeadersEntries: [string, string][] = [
+      ...fetcherOptsHeaders.entries(),
+    ];
+    // todo: If there are headers with multiple values, they will be lost. Is this a problem?
```

Comment:
I am not familiar with `graphql-ws` and have not investigated what it does with these key-values yet.

Can someone advise what the correct logic would be here? I am currently apply a lossy approach, but I do not know if that is ok here.

Reply:
I discovered that same-named headers are eagerly joined by Headers instance, so this is a non issue.